### PR TITLE
Get all coupons

### PR DESCRIPTION
**Pull request by shivamvijaywargi**
Fetch all coupons, do not merge as others are pending, keep checking the codes and mark the files as viewed we will merge this later in a go

### CODE REVIEW

File: tests/coupons/findAll.test.ts
Comment (sub1120) [inline tests/coupons/findAll.test.ts:1]: Shivam, is it possible to keep all coupon related test in one file. Like `coupon.test.ts` will contain all test cases related to coupons.

File: src/models/coupon.model.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  },
+  discountPercentage: {
+    type: Number,
+    required: [true, "Discount percentage is required"],
+    min: [0, "Discount percentage must be atleast 0"],
+    max: [100, "Discount percentage cannot be more than 100"],
+  },
+});
+
+const Coupon = model<ICoupon>("Coupon", couponSchema);
```
Comment (sub1120) [inline src/models/coupon.model.ts:25]: Can we rename this variable to `CouponModel`, same as we did for other Models.

File: src/models/coupon.model.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    required: [true, "Coupon code is required"],
+    minlength: [6, "Coupon code must be at least 6 characters"],
+    maxlength: [25, "Coupon code cannot exceed 25 characters"],
+    uppercase: true,
+    trim: true,
+  },
+  discountPercentage: {
+    type: Number,
+    required: [true, "Discount percentage is required"],
+    min: [0, "Discount percentage must be atleast 0"],
```
Comment (sub1120) [inline src/models/coupon.model.ts:20]: Shivam, for discount % the minimum value should be atleast 1, right? In validation also we have given so.

Comment (shivamvijaywargi) [reply src/models/coupon.model.ts:20]: Yes, I will update it.

File: tests/coupons/findAll.test.ts
Comment (shivamvijaywargi) [reply tests/coupons/findAll.test.ts:1]: Yes, that can also be done. Not a problem

File: src/models/coupon.model.ts (hunk 1 above)
Comment (shivamvijaywargi) [reply src/models/coupon.model.ts:25]: Oh, I might have missed it, bad habit. Definitely I will do that.

**Review by sub1120 — APPROVED** [review]

**Pull request merged**